### PR TITLE
Added support for '.' in Bucket Name

### PR DESCRIPTION
**Pull request by avinas-kumar**
Dear DIL maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [JIRA](https://jira01.corp.linkedin.com:8443/issues) issues and references them in the PR title. 

### Description
- [x] Here are some details about my PR, including screenshots (if applicable): Currently for the separation of bucket name and endpoint for s3 uploads we rely on '.' present in the host url. The left side of '.' is considered as bucket name and right side of it is considered an endpoint.
So added a new parameter in ms.source.s3.parameters for bucket name as "bucket_name" which would help to navigate through the edge cases. 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: Added UTs for it.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java [hunk 1]
```diff
@@ -130,4 +135,33 @@ private String getEndpointFromHost(String host) {
...
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the endpoint name without the bucket name
+   */
+  @VisibleForTesting
+  protected String getEndpoint(JsonObject parameters, String host) {
+    if (parameters.has(KEY_BUCKET_NAME)) {
+      String bucketName = parameters.get(KEY_BUCKET_NAME).getAsString().toLowerCase();
+      host = host.toLowerCase(Locale.ROOT).replace(bucketName, "");
```
Comment (dihu-linkedin) [inline cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java:163]: minor concern butt I don't see the tolowercase used in the original getEndpointFromHost method. Is there an existing use case that has issues with cases?

File: cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java [hunk 2]
```diff
@@ -130,4 +135,33 @@ private String getEndpointFromHost(String host) {
...
+      return parameters.get(KEY_BUCKET_NAME).getAsString();
+    }
+    return host.split("\\.")[0];
+  }
+
+  /**
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the endpoint name without the bucket name
```
Comment (dihu-linkedin) [inline cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java:157]: nit: in the comment maybe add some details to explain its difference from getEndpointFromHost. That will help with debugging in the future

File: cdi-core/src/test/java/com/linkedin/cdi/source/S3SourceV2Test.java [hunk 3]
```diff
@@ -36,4 +37,45 @@ public void testInitialization() throws UnsupportedEncodingException {
...
+  }
+
+  @Test
+  public void testEndpoint() {
+    S3SourceV2 s3SourceV2 = new S3SourceV2();
+    JsonObject parameteres =
+        new Gson().fromJson("{\"region\" : \"us-east-2\", \"bucket_name\" : \"colleCt-us-west-2.tealium.com\"}",
+            JsonObject.class);
+    String host = "collect-us-west-2.tealium.com.s3.amazonaws.com";
+    String endpoint = s3SourceV2.getEndpoint(parameteres, host);
```
Comment (dihu-linkedin) [inline cdi-core/src/test/java/com/linkedin/cdi/source/S3SourceV2Test.java:68]: nit: parameteres -> parameters

File: cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java (hunk 1 above)
Comment (avinas-kumar) [reply cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java:163]: We are replacing bucket name from the host, if some client enters bucket name in different case as in the source uri. It won't be replaced properly.

File: cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java (hunk 2 above)
Comment (avinas-kumar) [reply cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java:157]: Sure, I will add

File: cdi-core/src/test/java/com/linkedin/cdi/source/S3SourceV2Test.java (hunk 3 above)
Comment (avinas-kumar) [reply cdi-core/src/test/java/com/linkedin/cdi/source/S3SourceV2Test.java:68]: Sure with fix the typo.

**Review by dihu-linkedin — APPROVED** [review]
